client.go

```diff
@@ -111,7 +113,16 @@ func NewClient(zkquorum string) *Client {
 
 // CheckTable returns an error if the given table name doesn't exist.
 func (c *Client) CheckTable(table string) (*pb.GetResponse, error) {
-	resp, err := c.sendRpcToRegion(hrpc.NewGetStr(table, "theKey"))
+	resp, err := c.sendRpcToRegion(hrpc.NewGetStr(table, "theKey", nil))
+	if err != nil {
+		return nil, err
+	}
+	return resp.(*pb.GetResponse), err
+}
+
+// GetRow returns a single row fetched from hbase
+func (c *Client) GetRow(table string, rowkey string, families map[string][]string) (*pb.GetResponse, error) {
```

Comment:
My current solution relies on using those as keys to a map, and Go does not allow using slices as keys in maps. I'll figure out something else tomorrow so we can accept tables and rowkeys as []byte. 



> On Jun 8, 2015, at 21:34, Benoit Sigoure <notifications@github.com> wrote:
> 
> In client.go:
> 
> > @@ -111,7 +113,16 @@ func NewClient(zkquorum string) *Client {
> >  
> >  // CheckTable returns an error if the given table name doesn't exist.
> >  func (c *Client) CheckTable(table string) (*pb.GetResponse, error) {
> > -	resp, err := c.sendRpcToRegion(hrpc.NewGetStr(table, "theKey"))
> > +	resp, err := c.sendRpcToRegion(hrpc.NewGetStr(table, "theKey", nil))
> > +	if err != nil {
> > +		return nil, err
> > +	}
> > +	return resp.(*pb.GetResponse), err
> > +}
> > +
> > +// GetRow returns a single row fetched from hbase
> > +func (c *Client) GetRow(table string, rowkey string, families map[string][]string) (*pb.GetResponse, error) {
> Not for later: we should have an API that accepts mostly []byte, but it's also nice to provide an additional function that accepts string arguments, because it's a common thing to use too.
> 
> ―
> Reply to this email directly or view it on GitHub.
>